src/index.js

```diff
@@ -57,14 +57,49 @@ class Storage {
           if (db && db.status === 'unknown-shim') db.status = 'open'
           resolve(store)
         })
-      } else {
-        // Default leveldown or level-js store with directory creation
-        if (fs && fs.mkdirSync) fs.mkdirSync(directory, { recursive: true })
-        const db = level(directory, options)
-        await db.open()
-        resolve(db)
-      }
+      })
+    } else {
+      // Default leveldown or level-js store with directory creation
+      if (fs && fs.mkdirSync) fs.mkdirSync(directory, { recursive: true })
+      db = level(directory, options)
+      await db.open()
+    }
+
+    store = store || db
+    const leveljs = reachdown(store, 'level-js', true)
+    if (leveljs && leveljs.upgrade) {
+      await this._leveljsUpgrade(store, leveljs)
+    }
+
+    return store
+  }
+
+  // upgrade level-js to version 5.0.0
+  // https://github.com/Level/level-js/blob/master/UPGRADING.md#500
+  async _leveljsUpgrade (store, leveljs) {
+    const upgradeKey = new Uint8Array([0])
+
+    async function isUpgraded () {
+      return Boolean(await store.get(upgradeKey).catch(e => false))
+    }
+    async function setUpgraded () {
+      await store.put(upgradeKey, '1')
+    }
+
+    // return if already upgraded
+    if (await isUpgraded()) {
+      store._leveljs5 = true
```

Comment:
let's change the tests to read from the db directly rather than this stray object property